src/vfs_glusterfs.c

```diff
@@ -144,6 +144,9 @@ static void glfs_clear_preopened(glfs_t *fs)
 			DLIST_REMOVE(glfs_preopened, entry);
 
 			glfs_fini(entry->fs);
+			
+			talloc_free(entry->connectpath);
+			talloc_free(entry->volume);
```

Comment:
This would serve as the best example of why Gluster-Samba team appreciates talloc. Talloc library says you don't need to call free on each node of the tree if you are calling free on the parent, so talloc_free(entry) cleans up connectpath and volume too.